src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -1300,16 +1299,15 @@ private static List<Row> readWithPaxos(List<ReadCommand> commands, ConsistencyLe
         final ConsistencyLevel consistencyForCommit = consistencyLevel == ConsistencyLevel.LOCAL_SERIAL
                                                                         ? ConsistencyLevel.LOCAL_QUORUM
                                                                         : ConsistencyLevel.QUORUM;
-        final ConsistencyLevel consistencyForFetch = maybeCoerceReadConsistencyLevel(consistencyForCommit);
-        ClientRequestMetrics readMetrics = consistencyLevelReadMetrics.get(consistencyForFetch);
+        ReadCommand command = commands.get(0);
+        consistencyLevel = maybeCoerceReadConsistencyLevel(command.ksName, consistencyForCommit);
```

Comment:
Do we have the guarantee that all read commands in the list use the same keyspace?

Reply:
If not, and we wanna keep it simple, you probably wanna default to the highest consistency levels for all read commands, ie do a "if any keyspace is coerced"

Reply:
This code path in particular is a bit weird, where we'll throw if we actually have [more than one command](https://github.com/palantir/cassandra/pull/679/files#diff-4f6615c703489b87cafd23c0dc5a4edc4af27c33aba172870b30e38a7d8faa1dR1309). Agreed that it feels a bit dangerous, so could drop a comment that this is only safe because of that check/if that ever changed, we'd need to update this as well.